arith/src/extension_field/gf2_128/neon.rs

```diff
@@ -176,27 +176,26 @@ impl BinomialExtensionField for NeonGF2_128 {
 
     #[inline(always)]
     fn mul_by_base_field(&self, base: &Self::BaseField) -> Self {
-        if base.v == 0 {
+        if base.is_zero() {
             Self::zero()
         } else {
             *self
         }
     }
 
     #[inline(always)]
-    fn add_by_base_field(&self, _base: &Self::BaseField) -> Self {
-        todo!()
-        // let mut res = *self;
-        // res.v = unsafe { _mm_xor_si128(res.v, _mm_set_epi64x(0, base.v as i64)) };
-        // res
+    fn add_by_base_field(&self, base: &Self::BaseField) -> Self {
+        if base.is_zero() {
+            return *self;
+        }
+        add_internal(&Self::one(), self)
     }
 
     #[inline(always)]
     fn first_base_field(&self) -> Self::BaseField {
```

Comment:
We should remove this function

Reply:
wdym? take away the trait method together with all the implementations for `first_base_field`?